File: 3a.py
#! /usr/bin/python3
import sys

bit_frequency = []
with open(sys.argv[1], 'r') as fh:
    for line in fh:
        for i in range(0, len(line) - 1):
            if len(bit_frequency) <= i:
                bit_frequency.append([0, 0])
            bit_frequency[i][int(line[i])] += 1

gamma = "".join(map(lambda x: "0" if x[0] > x[1] else "1", bit_frequency))
epsilon = gamma.translate(gamma.maketrans("01", "10"))
print(int(gamma, 2) * int(epsilon, 2))

File: 3a.pl
#!/usr/bin/perl
use warnings;
use strict;
use feature qw{ say };
use Syntax::Construct qw{ /r };

use ARGV::OrDATA;

my @bit_frequency;
while (<>) {
    chomp;
    my $i = 0;
    ++$bit_frequency[$i++][$_] for split //;
}
my $gamma = join "", map $_->[0] > $_->[1] ? 0 : 1, @bit_frequency;
my $epsilon = $gamma =~ tr/01/10/r;
say oct("0b$gamma") * oct("0b$epsilon");

__DATA__
00100
11110
10110
10111
10101
01111
00111
11100
10000
11001
00010
01010
